src/overlays/actors/ovl_En_Time_Tag/z_en_time_tag.c

```diff
@@ -42,40 +42,268 @@ const ActorInit En_Time_Tag_InitVars = {
+        this->actor.home.rot.x = 5;
+        this->actionFunc = func_80ACA348;
+        globalCtx->msgCtx.unk11F10 = 0;
+        globalCtx->msgCtx.msgMode = 0;
+    }
+}
+
+void func_80ACA418(EnTimeTag* this, GlobalContext* globalCtx) {
+    switch (Message_GetState(&globalCtx->msgCtx)) {
+        case 5:
+            if (Message_ShouldAdvance(globalCtx)) {
+                switch (globalCtx->msgCtx.currentTextId) {
+                    case 0x101C:
+                    case 0x101D:
+                    case 0x101E:
+                    case 0x122D:
+                        func_80151938(globalCtx, globalCtx->msgCtx.currentTextId + 1);
+                        break;
+
+                    case 0x101F:
+                    case 0x122A:
+                    case 0x1230:
+                        func_801477B4(globalCtx);
+                        this->actionFunc = func_80ACA5F8;
+                        if (ActorCutscene_GetCurrentIndex() == this->actor.cutscene) {
+                            ActorCutscene_Stop(this->actor.cutscene);
+                        }
+                        break;
+
+                    case 0x122B:
+                        func_80152434(globalCtx, 0x3F);
+                        this->actionFunc = func_80ACA3C0;
+                        this->actor.home.rot.z = 0;
+                        break;
+
+                    case 0x122E:
+                        func_80152434(globalCtx, 0x40);
+                        this->actionFunc = func_80ACA3C0;
+                        this->actor.home.rot.z = 1;
+                        break;
+                }
+            }
+            break;
+
+        case 2:
+            this->actionFunc = func_80ACA5F8;
+            break;
+    }
+
+    if (this->actor.home.rot.x != 0) {
+        if (this->actor.cutscene == -1) {
+            this->actor.home.rot.x = 0;
+        } else if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+            ActorCutscene_Start(this->actor.cutscene, &this->actor);
+            this->actor.home.rot.x = 0;
+        } else {
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        }
+    }
+}
+
+void func_80ACA5F8(EnTimeTag* this, GlobalContext* globalCtx) {
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        if (gSaveContext.save.playerForm == PLAYER_FORM_ZORA) {
+            if ((ENTIMETAG_GET_SWITCHFLAG(&this->actor)) == 1) {
+                Message_StartTextbox(globalCtx, 0x101C, &this->actor);
+            } else {
+                Message_StartTextbox(globalCtx, 0x122B, &this->actor);
+            }
+            this->actor.home.rot.x = 1;
+        } else {
+            Message_StartTextbox(globalCtx, 0x122A, &this->actor);
+            if (0) {}
+            ((EnElf*)GET_PLAYER(globalCtx)->tatlActor)->unk_264 |= 4;
```

Comment:
Is it possible to make `tatlActor` an `EnElf*` instead of casting?

Reply:
I don't know how to forward-declare it and all that.